### PR TITLE
Azure configuration via ENV variables

### PR DESCRIPTION
**Pull request by petems**
* Unlike the AWS library, this needs to be done with logic
* Closes https://github.com/hashicorp/go-discover/issues/96

### CODE REVIEW

File: provider/azure/azure_discover.go [hunk 1]
```diff
@@ -28,7 +29,16 @@ func (p *Provider) Help() string {
    tenant_id:         The id of the tenant
    client_id:         The id of the client
    subscription_id:   The id of the subscription
-   secret_access_key: The authentication credential
+	 secret_access_key: The authentication credential
```
Comment (i0rek) [inline provider/azure/azure_discover.go]: Somehow the indentation is different than the other things. Maybe tabs vs spaces?

File: provider/azure/azure_discover.go [hunk 2]
```diff
@@ -40,14 +50,22 @@ func (p *Provider) Help() string {
    resource_group:    The name of the resource group to filter on
    vm_scale_set:      The name of the virtual machine scale set to filter on
 
-   When using tags the only permission needed is the 'ListAll' method for
-   'NetworkInterfaces'. When using Virtual Machine Scale Sets the only Role
-   Action needed is 'Microsoft.Compute/virtualMachineScaleSets/*/read'.
+   When using tags the only permission needed is Microsoft.Network/networkInterfaces/*
+
+	 When using Virtual Machine Scale Sets the only role action needed is Microsoft.Compute/virtualMachineScaleSets/*/read.
```
Comment (i0rek) [inline provider/azure/azure_discover.go]: indentation again.

File: provider/azure/azure_discover.go [hunk 3]
```diff
@@ -40,14 +50,22 @@ func (p *Provider) Help() string {
...
 
    It is recommended you make a dedicated key used only for auto-joining.
 `
 }
 
+// argsOrEnv allows you to pick an environmental variable for a setting if the arg is not set
+func argsOrEnv(args map[string]string, key, env string) string {
+	if value := args[key]; value != "" {
```
Comment (i0rek) [inline provider/azure/azure_discover.go]: you could also check the second returned value: `if value, ok := args[key]; ok {`

File: provider/azure/azure_discover.go (hunk 1 above)
Comment (petems) [reply provider/azure/azure_discover.go]: Fixed! 👍

File: provider/azure/azure_discover.go (hunk 2 above)
Comment (petems) [reply provider/azure/azure_discover.go]: Fixed! 👍

File: provider/azure/azure_discover.go (hunk 3 above)
Comment (petems) [reply provider/azure/azure_discover.go]: Fixed! 👍

**Comment by petems** [comment]
Fixed issues, merging after green! 👍

**Pull request merged**